tests/framework/extensions/hardening/k3s/harden_nodes.go

```diff
@@ -0,0 +1,79 @@
+package hardening
+
+import (
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/pkg/nodes"
+	"github.com/sirupsen/logrus"
+)
+
+func CheckClusterHardening(client *rancher.Client, hardened bool, nodes []*nodes.Node, nodeRoles []string) error {
+	for key, node := range nodes {
+		logrus.Infof("Setting kernel parameters on node %s", node.NodeID)
+		_, err := node.ExecuteCommand("sudo setenforce 1")
+		if err != nil {
+			return err
+		}
+		_, err = node.ExecuteCommand("sudo bash -c 'echo vm.panic_on_oom=0 >> /etc/sysctl.conf'")
+		if err != nil {
+			return err
+		}
+		_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.panic=10 >> /etc/sysctl.conf'")
+		if err != nil {
+			return err
+		}
+		_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.panic_on_oops=1 >> /etc/sysctl.conf'")
+		if err != nil {
+			return err
+		}
+		_, err = node.ExecuteCommand("sudo bash -c 'echo kernel.keys.root_maxbytes=25000000 >> /etc/sysctl.conf'")
+		if err != nil {
+			return err
+		}
+		_, err = node.ExecuteCommand("sudo bash -c 'sysctl -p /etc/sysctl.conf'")
+		if err != nil {
+			return err
+		}
+
+		if nodeRoles[key] == "--etcd --controlplane --worker" || nodeRoles[key] == " --controlplane" {
```

Comment:
There's a space in the value, should be `--controlplane`. 
`nodeRoles[key] == " --controlplane"`

Reply:
Yeah, it's actually supposed to have the space funny enough. Since in the `custom_cluster_test.go` test, there is a space in each of the node roles in the dynamic input. So if you don't have a space in this conditional, then it's going to fail.